src/database/firebaseAuthAccess.ts

```diff
@@ -0,0 +1,100 @@
+import { AccessDatabase } from "../types/users";
+import { URL } from "./firebase";
+import { headers } from "./firebase";
+
+export const saveDatabaseIWantToAccess = async (
+  userIWantToAccess: string,
+  user: string | null | undefined,
+  token: string | undefined
+) => {
+  let message = {};
+
+  await fetch(`${URL}authaccess`, {
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify({ userIWantToAccess, user, token }),
+  })
+    .then((resp) => resp.json())
+    .then((resp) => {
+      if (resp.error) {
+        message = { error: resp.error };
+      }
+    });
+
+  return message;
```

Comment:
Sugestão: Acho que você consegue deixar o código mais simples se fizer isso...

```suggestion
  const resp = await fetch(`${URL}authaccess`, {
    method: "POST",
    headers: headers,
    body: JSON.stringify({ userIWantToAccess, user, token }),
  });
  if (resp.json().error) {
      message = { error: resp.error };
  }
  return message;
```

Testa, por favor, mas deveria funcionar.  Aí você pode simplificar as outras APIs também.